### PR TITLE
[Merged by Bors] - chore: deprecate `UniformSpace.Completion.Continuous.mul`

### PR DESCRIPTION
**Pull request by eric-wieser**
By adding a missing instance, this becomes just a really weird way to spell `Continuous.mul`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Algebra/UniformRing.lean [hunk 1]
```diff
@@ -65,63 +65,66 @@ theorem coe_mul (a b : α) : ((a * b : α) : Completion α) = a * b :=
...
-  Continuous.comp continuous_mul (Continuous.prod_mk hf hg : _)
+  hf.mul hg
 
 instance ring : Ring (Completion α) :=
   { AddMonoidWithOne.unary, (inferInstanceAs (AddCommGroup (Completion α))),
       (inferInstanceAs (Mul (Completion α))), (inferInstanceAs (One (Completion α))) with
     zero_mul := fun a =>
       Completion.induction_on a
-        (isClosed_eq (Continuous.mul continuous_const continuous_id) continuous_const)
+        (isClosed_eq (continuous_const.mul continuous_id) continuous_const)
```
Comment (Ruben-VandeVelde) [inline Mathlib/Topology/Algebra/UniformRing.lean:89]: Can `fun_prop` solve these now?

**Review by Ruben-VandeVelde — APPROVED** [review]
Makes sense to me

maintainer merge

**Comment by jcommelin** [comment]
bors d+

**Comment by PatrickMassot** [comment]
Yes this was very old code ported without change.

File: Mathlib/Topology/Algebra/UniformRing.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Topology/Algebra/UniformRing.lean:89]: Maybe, but that feels out of scope for a deprecation. Probably `continuity` is enough here, but `by continuity` isn't much shorter than `continuous_const.mul continuous_id`, is unlikely to be faster.

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**